gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -109,10 +109,13 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
         assert(kotlinCompileProvider.name.startsWith("compile"))
         val kspTaskName = kotlinCompileProvider.name.replaceFirst("compile", "ksp")
         val destinationDir = File(project.buildDir, "generated/ksp")
```

Comment:
i think the problem is here: ` val destinationDir = File(project.buildDir, "generated/ksp")`
it does not create the diretory specific to the compilation, instead it is global for KSP.
we should probably get another identifier from the kotlin compilation to disguish them from eachother.

Reply:
Yes, that is indeed true. The challenge is to set it to something reasonable. Currently, there are two possible trees under `generated/ksp`: `src/$sourceSetName` and `classes/$sourceSetName`. (Of which it looks like the latter is never used, because the subsequent `compileKotlin` tasks create the .class files in `build/classes`). So even though it is most logical, unfortunately it is not as simple as appending the sourceset name to `generated/ksp`. Then you would get `generated/ksp/test/src/test` and `generated/ksp/test/classes/test`. 

Additionally, in this line all dirs specific to the sourceset are also set as outputs: 
https://github.com/google/ksp/blob/e893da3c09c53e1e7876fa62aea1be8ae6c619a5/gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt#L121

@yigit Any suggestions? Happy to implement and test, but not sure how to proceed.